pkg/abi/abidecode.go

```diff
@@ -88,12 +88,18 @@ func decodeABIElement(ctx context.Context, breadcrumbs string, block []byte, hea
 		}
 		return 32, cv, err
 	case TupleComponent:
-		headOffset, err := decodeABILength(ctx, breadcrumbs, block, headPosition)
+		dynamic, err := isDynamicType(ctx, component)
 		if err != nil {
 			return -1, nil, err
 		}
-		headStart += headOffset
-		headPosition = headStart
+		if dynamic {
+			headOffset, err := decodeABILength(ctx, breadcrumbs, block, headPosition)
+			if err != nil {
+				return -1, nil, err
+			}
+			headStart += headOffset
+			headPosition = headStart
+		}
 		_, cv, err := walkDynamicChildArrayABIBytes(ctx, "tup", breadcrumbs, block, headStart, headPosition, component, component.tupleChildren)
```

Comment:
How does the caller know how many head bytes were ready @EnriqueL8 ?

Think given ^^ it might be worth adding a test case that has a nested static tuple, inside of a dynamic tuple, with more items after it in the dynamic one.

`(string, (address , uint256), string, uint256)` for example